go-controller/pkg/ovn/egressip.go

```diff
@@ -1107,6 +1107,11 @@ func (oc *DefaultNetworkController) addPodEgressIPAssignments(name string, statu
 	if err != nil {
 		return nil
 	}
+	// If pod is already in succeeded or failed state, return it without proceeding further.
+	if util.PodCompleted(pod) {
```

Comment:
aren't completed pods deleted from the logical port cache when they do into the completed state?
is this the case of stateful sets and them being completed?

Reply:
@tssurya This is done based on pod add event handling done for completed case here https://github.com/openshift/ovn-kubernetes/blob/release-4.10/go-controller/pkg/ovn/ovn.go#L625, so thought there would be a race condition for logical port cache cleanup and adding a check here makes sense.

Reply:
yes @tssurya , we noticed stale snat and lrp entries on a cluster, believe this is due to 'completed' pods. so did this based on code study, we would be trying out this [fix](https://github.com/openshift/ovn-kubernetes/pull/1658) on a test cluster today. based on test outcome, will amend suggested changes in this PR.

Reply:
I think you should move this check to the top of the function.

Reply:
now tests are looking promising in QE lab, will add a required test to cover 'completed' pod case.

Reply:
tests are added now.